### ISSUE TITLE
Bug: ec2 cluster with Amazon Linux images fail to start

### ISSUE THREAD

**Issue by patrick0057** [rancher/rancher#21648]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1) Create new Amazon EC2 cluster in us-east1
2) Use one of the following AWS Linux distro ami's:
ami-0b898040803850657
ami-0f2e8e5663e16b436

**Result:**
VPS won't boot, it will start in a stopped state.  If you take an instance screenshot at the right time during a startup attempt, you will see the following error.
```
Booting from Hard Disk...
Boot from Hard Disk failed: not a bootable disk

No bootable device.
Powering off in 30 seconds.
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):v2.2.3
- Installation option (single install/HA):HA

gz#13285

**Comment by patrick0057** [comment]
This appears to be a bug in docker-machine.

https://github.com/docker/machine/issues/4738

**Comment by dkeightley** [comment]
For reference adding another error reported during node bootstrap:

`Error detecting OS: OS type not recognized`

**Comment by sowmyav27** [comment]
2.5 PR: https://github.com/rancher/rancher/pull/30368

**Comment by sadiapoddar** [comment]
Tried to reproduce with rancher v2.2.3
Steps
1. created ec2 template for the node driver by providing ami as ami-0b898040803850657
2. created a cluster
3. the cluster didn't come up
4. from asw console seeing all the vms created by the node driver in stopped state and no IP
Don't have all the boot error message, since we started the VM creation from rancher UI.


Verified the fix with the latest master e4ce376 and it's not working 

1. created ec2 template for the node driver by providing ami as ami-0b898040803850657
2. created a cluster
3. the cluster didn't come up
4. from aws console, seeing all the vms are up and running 
5. rancher UI showing machine creating error to detect os 

<img width="1368" alt="Screen Shot 2020-12-09 at 6 22 35 PM" src="https://user-images.githubusercontent.com/72409237/101713384-69b00180-3a4c-11eb-9754-5ea4f94207a5.png">

6. logs from rancher server 

20/12/10 02:15:32 [INFO] [node-controller-rancher-machine] Running pre-create checks...
2020/12/10 02:15:33 [INFO] [node-controller-rancher-machine] Creating machine...
2020/12/10 02:15:33 [INFO] [node-controller-rancher-machine] (sp.x.x.x) Launching instance...
2020/12/10 02:15:33 [INFO] [node-controller-rancher-machine] (x.x.x) Creating New SSH Key
2020/12/10 02:15:33 [INFO] [node-controller-rancher-machine] (x.x.x.x) creating key pair: sp-etcd-cnt-wk1-slYfW
2020/12/10 02:15:34 [INFO] [node-controller-rancher-machine] (xxxxxx) configuring security groups in vpc-c66fb6bc
2020/12/10 02:15:35 [INFO] [node-controller-rancher-machine] (xxxx) found existing security group (all-open) in vpc-c66fb6bc
2020/12/10 02:15:35 [INFO] [node-controller-rancher-machine] (xxx) configuring security group authorization for 0.0.0.0/0
2020/12/10 02:15:35 [INFO] [node-controller-rancher-machine] (xx) launching instance in subnet subnet-8a8308a4
2020/12/10 02:15:36 [INFO] [node-controller-rancher-machine] (xxx) waiting for ip address to become available
2020/12/10 02:15:37 [INFO] [node-controller-rancher-machine] (xx1) InvalidInstanceID.NotFound: The instance ID 'i-0bc4d467ece857a56' does not exist
2020/12/10 02:15:37 [INFO] [node-controller-rancher-machine] (xxx1) 	status code: 400, request id xxxx
2020/12/10 02:15:41 [INFO] [node-controller-rancher-machine] (x) Got the IP Address, it's "x.x.x.x"
2020/12/10 02:16:13 [INFO] [node-controller-rancher-machine] (x) created instance ID xxxxx, IP address x.x.x.x, Private IP address x.x.x.x
2020/12/10 02:16:13 [INFO] [node-controller-rancher-machine] (x1) Settings tags for instance
2020/12/10 02:16:13 [INFO] [node-controller-rancher-machine] Waiting for machine to be running, this may take a few minutes...
2020/12/10 02:16:14 [INFO] [node-controller-rancher-machine] Detecting operating system of created instance...
2020/12/10 02:16:14 [INFO] [node-controller-rancher-machine] Waiting for SSH to be available...
2020-12-10 02:17:15.817593 I | mvcc: store.index: compact 14302
2020-12-10 02:17:15.859704 I | mvcc: finished scheduled compaction at 14302 (took 39.351494ms)

**Comment by maggieliu** [comment]
@sadiapoddar Can you try it with 2.5?

**Comment by sadiapoddar** [comment]
@maggieliu Sure I will do

**Comment by sadiapoddar** [comment]
Tested the fix on master-e4ce376 and v2.5-3463f5a
Used the amazon ami-0b898040803850657. 
When users will use this ami from to create node-template, the user has to be
"ec2-user". The fix is working on both branches. 

Steps
1. created a node template using the amazon linux ami, mi-0b898040803850657
2. created a cluster using the node-template
3. the cluster is up and running

closing the bug.
